subsys/net/lib/coap/coap_link_format.c

```diff
@@ -526,76 +476,64 @@ int _coap_well_known_core_get(struct coap_resource *resource,
 
 #else
 
-static int format_uri(const char * const *path, struct net_buf *buf)
+static int format_uri(const char * const *path, struct net_pkt *pkt)
 {
-	static const char prefix[] = "</";
 	const char * const *p;
-	char *str;
+	char *prefix = "</";
 
 	if (!path) {
 		return -EINVAL;
 	}
 
-	str = net_buf_add(buf, sizeof(prefix) - 1);
-	strncpy(str, prefix, sizeof(prefix) - 1);
+	net_pkt_append_all(pkt, strlen(prefix), (u8_t *) prefix, K_FOREVER);
```

Comment:
Same here, and naturally we should check the return value here and in previous places.